src/inflators/spot-light.ts

```diff
@@ -0,0 +1,53 @@
+import { addComponent } from "bitecs";
+import { addObject3DComponent } from "../utils/jsx-entity";
+import { SpotLightTag, LightTag } from "../bit-components";
+import { SpotLight } from "three";
+import { HubsWorld } from "../app";
+
+export type SpotLightParams = {
+  color: string;
+  intensity: number;
+  range: number;
+  decay: number;
+  innerConeAngle: number;
+  outerConeAngle: number;
+  castShadow: boolean;
+  shadowMapResolution: [number, number];
+  shadowBias: number;
+  shadowRadius: number;
+};
+
+const DEFAULTS = {
+  intensity: 1.0,
+  range: 0,
+  decay: 2.0,
+  innerConeAngle: 0,
+  outerConeAngle: Math.PI / 4.0,
+  castShadow: true,
+  shadowMapResolution: [512, 512],
+  shadowBias: 0,
+  shadowRadius: 1.0
+};
+
+export function inflateSpotLight(world: HubsWorld, eid: number, params: SpotLightParams) {
```

Comment:
This was discussed in anotehr PR, so whatever we decide there should apply here too... Since we provide default values the properties passed to the inflator (and exposed to JSX) should be optional, but then the type after applying defaults should have all fields be required. We can do that with typescript Partial/Required utility types.